File: Data Structures & Algorithms/car-fleet/submission-2.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int fleets = 0;
        List<Integer> ind = IntStream.range(0, position.length)
        .boxed()
        .sorted((a, b) -> Integer.compare(position[b], position[a]))
        .collect(Collectors.toList());

        double prevTime = 0;
        for (int i = 0; i < position.length; i++) {
            double newTime = (double) (target - position[ind.get(i)]) / speed[ind.get(i)];
            if (newTime > prevTime) {
                fleets += 1;
                prevTime = newTime;
            }
        }
        return fleets;
    }
}

File: Data Structures & Algorithms/count-subsequences/submission-2.py
class Solution:
    def numDistinct(self, s: str, t: str) -> int:
        n, m = len(s), len(t)

        dp = [[0 for j in range(m+1)] for i in range(n+1)]

        j = 0
        for i in range(n+1):
            dp[i][j] = 1
            
        for i in range(1, n+1):
            for j in range(1, m+1):
                ntake = dp[i-1][j]
                take = 0
                if s[i-1] == t[j-1]:
                    take = dp[i-1][j-1]
                dp[i][j] = take + ntake
        
        return dp[n][m]